src/project/project-actions.tsx

```diff
@@ -476,18 +554,50 @@ export class ProjectActions {
     return this.fs.setProjectName(name);
   };
 
-  private handleWebUSBError(e: any) {
+  private async handleNotFound() {
+    // Temporary hide for French language users.
+    if (this.settings.values.languageId !== "en") {
+      return;
+    }
+    const choice = await this.dialogs.show<NotFoundChoice>((callback) => (
+      <NotFoundDialog callback={callback} />
+    ));
+    switch (choice) {
+      case NotFoundChoice.Retry: {
+        this.connectInternal();
+        return;
+      }
+      case NotFoundChoice.ReviewDevice: {
+        this.connect(true);
+        return;
+      }
+    }
+  }
+
+  private async handleWebUSBError(e: any) {
     if (e instanceof WebUSBError) {
       switch (e.code) {
         case "no-device-selected": {
-          // User just cancelled the browser dialog so no further response needed.
+          // User just cancelled the browser dialog, perhaps because there
+          // where no devices.
+          await this.handleNotFound();
           return;
         }
         case "device-disconnected": {
           // The UI will already reflect the disconnection.
           return;
         }
         case "update-req":
+          this.device.clearDevice();
+          // Temporary hide for French language users.
+          if (this.settings.values.languageId !== "en") {
+            return this.actionFeedback.expectedError(
+              this.webusbErrorMessage(e.code)
+            );
+          }
+          return await this.dialogs.show<void>((callback) => (
+            <FirmwareDialog callback={callback} />
+          ));
```

Comment:
Let's pull this out into a method similar to handleNotFound for consistency.

Reply:
[`f012147`](https://github.com/microbit-foundation/python-editor-next/pull/702/commits/f0121472c8ef006e93cb31b5f9f675aab6d73f65)